.github/workflows/python-app.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```

Comment:
Very minor comment.

this is ok... as best practice we try to "pin" as many dependecies of the test environments.

With `latest` it means that we will get an upgrade from 18.04 to 20.04 or a future 22.04 whener GitHub decides to do .

For this project is ok... for more stable project we want a greater control.

This is why we don't just install any flake8 version, but install a specific version.

So for the testing environment we pin anything. This will prevent the test from "automatically" failing in the future if flake8 or other tools are generating new errors.